cmds/legacy_boot/legacy_boot.go

```diff
@@ -0,0 +1,383 @@
+		}
+	}
+	return_value = false
+	return return_value
+}
+
+func umount_entry(path string) bool {
+	var return_value bool
+	var flags int
+	// Was supposed to be unecessary for kernel 4.x.x
+	flags = syscall.MNT_DETACH
+	err := syscall.Unmount(path, flags)
+	if err == nil {
+		return true
+	}
+	return_value = false
+	return return_value
+}
+
+// This function is looking for grub.cfg file
+// and return absolute path to it
+
+func check_boot_entry(mount_point string) string {
+	_, err := os.Stat(mount_point + "/boot")
+	if err == nil {
+		// The boot directory is there
+		_, err2 := os.Stat(mount_point + "/boot" + "/grub")
+		if err2 == nil {
+			_, err3 := os.Stat(mount_point + "/boot" + "/grub" + "/grub.cfg")
+			if err3 == nil {
+				// found
+				return (mount_point + "/boot" + "/grub")
+			}
+		}
+	}
+	return ""
+
+}
+
+// This function is parsing a grub.cfg file
+// input: absolute directory path to grub.cfg
+// output: Return a list of strings with the following format
+//	 line[3*x] - menuconfig
+//	 line[3*x+1] - linux kernel + boot options
+// 	 line[3*x+2] - initrd
+// and the default boot entry configured into grub.cfg
+
+func get_file_menu_content(path string) ([]string, int) {
+	var return_value []string
+	file, _ := os.Open(path + "/grub.cfg")
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	var status int
+	var int_return int
+	int_return = 0
+	status = 0
+	// When status = 0 we are looking for a menu entry
+	// When status = 1 we are looking for a linux entry
+	// When status = 2 we are looking for a initrd entry
+	var trimmed_line string
+	for scanner.Scan() {
+		trimmed_line = strings.TrimSpace(scanner.Text())
+		trimmed_line = strings.Join(strings.Fields(trimmed_line), " ")
+		if strings.HasPrefix(trimmed_line, "#") == false {
+			if (strings.HasPrefix(trimmed_line, "set default=") == true) && (status == 0) {
+				fmt.Sscanf(trimmed_line, "set default=\"%d\"", &int_return)
+			}
+			if (strings.HasPrefix(trimmed_line, "menuentry ") == true) && (status == 0) {
+				status = 1
+				return_value = append(return_value, trimmed_line)
+			}
+			if (strings.HasPrefix(trimmed_line, "linux ") == true) && (status == 1) {
+				status = 2
+				return_value = append(return_value, trimmed_line)
+			}
+			if (strings.HasPrefix(trimmed_line, "initrd ") == true) && (status == 2) {
+				status = 0
+				return_value = append(return_value, trimmed_line)
+			}
+		}
+	}
+	return return_value, int_return
+
+}
+
+func copy_local(path string) string {
+	var dest string
+	result := strings.Split(path, "/")
+	for _, entry := range result {
+		dest = entry
+	}
+	dest = "/tmp/" + dest
+	srcFile, err := os.Open(path)
+	check(err)
+	defer srcFile.Close()
+
+	destFile, err := os.Create(dest) // creates if file doesn't exist
+	check(err)
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, srcFile) // check first var for number of bytes copied
+	check(err)
+
+	err = destFile.Sync()
```

Comment:
Close syncs, doesn't it?

Reply:
Well,  ... depends on what you mean. In general, you can't count on that, so if you really want the bits to hit the disc, this is good practice I think.